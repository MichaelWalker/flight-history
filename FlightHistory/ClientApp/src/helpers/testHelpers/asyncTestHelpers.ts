import { act } from "@testing-library/react-hooks";

export async function runActionAndWaitForAsyncActionsToComplete(
    action: () => Promise<unknown> | void,
): Promise<void> {
    await act(async () => {
        await action();
    });
}

export async function quickResolve<T>(callback: () => T): Promise<T> {
    return new Promise((resolve) => {
        setTimeout(() => {
            resolve(callback());
        }, 0);
    });
}

export async function quickReject<T>(callback: () => T): Promise<T> {
    return new Promise((resolve, reject) => {
        setTimeout(() => {
            reject(callback());
        }, 0);
    });
}

export class WrappedPromise<T> {
    private readonly promise: Promise<T>;
    private resolved = false;
    private result: T | undefined;

    constructor(promise: Promise<T>) {
        this.promise = promise;

        promise.then((result) => {
            this.resolved = true;
            this.result = result;
        });
    }

    isResolved(): boolean {
        return this.resolved;
    }
}
