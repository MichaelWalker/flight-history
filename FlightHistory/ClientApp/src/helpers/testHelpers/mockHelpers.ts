import * as globals from "../globals";

export function mockUseSampleData(useSampleData: boolean) {
    jest.spyOn(globals, "useSampleData").mockReturnValue(useSampleData);
}

export function mockUseConsoleLogging(useConsoleLogging: boolean) {
    jest.spyOn(globals, "useConsoleLogging").mockReturnValue(useConsoleLogging);
}
