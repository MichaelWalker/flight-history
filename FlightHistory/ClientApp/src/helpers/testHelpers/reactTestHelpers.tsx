import type { ReactElement } from "react";
import React from "react";
import type { RenderResult } from "@testing-library/react";
import { render, screen, waitFor } from "@testing-library/react";
import { MemoryRouter } from "react-router-dom";

export function renderWithRouter(component: ReactElement): RenderResult {
    return render(<MemoryRouter>{component}</MemoryRouter>);
}

export async function waitForTableToHaveResults(): Promise<void> {
    await waitFor(() => expect(screen.getAllByRole("row").length).toBeGreaterThan(1));
}
