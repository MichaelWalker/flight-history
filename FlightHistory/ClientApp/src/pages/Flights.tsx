import React, {FunctionComponent} from "react";
import {Page} from "./Page";
import {FlightsTable} from "../components/flights/FlightsTable";

export const FlightsPage: FunctionComponent = () => {
    return (
        <Page title={"Flights"}>
            <FlightsTable/>
        </Page>
    );
};
