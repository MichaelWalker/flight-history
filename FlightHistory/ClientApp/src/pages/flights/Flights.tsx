import type { FunctionComponent } from "react";
import React from "react";
import { PrimaryLinkButton } from "../../components/buttons/linkButton/LinkButton";
import { Page } from "../Page";
import { FlightsTable } from "../../components/flights/FlightsTable";
import { CardSection } from "../../components/layouts/CardSection";

export const FlightsPage: FunctionComponent = () => {
    return (
        <Page title={"Flights"}>
            <CardSection
                title="Flights"
                actions={<PrimaryLinkButton to={"/flights/add"}>Add Flight</PrimaryLinkButton>}
            >
                <FlightsTable />
            </CardSection>
        </Page>
    );
};
