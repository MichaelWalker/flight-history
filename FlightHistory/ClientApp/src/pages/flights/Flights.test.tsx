import "isomorphic-fetch";
import React from "react";
import {
    renderWithRouter,
    waitForTableToHaveResults,
} from "../../helpers/testHelpers/reactTestHelpers";
import { mockUseConsoleLogging, mockUseSampleData } from "../../helpers/testHelpers/mockHelpers";
import { FlightsPage } from "./Flights";

describe("Flights Page", () => {
    it("matches snapshot", async () => {
        mockUseSampleData(true);
        mockUseConsoleLogging(false);

        const { container } = renderWithRouter(<FlightsPage />);

        await waitForTableToHaveResults();
        expect(container).toMatchSnapshot();
    });
});
