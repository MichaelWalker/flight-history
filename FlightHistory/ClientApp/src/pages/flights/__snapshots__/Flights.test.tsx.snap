// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Flights Page matches snapshot 1`] = `
<div>
  <div
    class="css-opzq46"
  >
    <div
      class="css-1tyka62"
    >
      <a
        class="css-dp7dt0"
        href="/"
      >
        Flight History
      </a>
      <nav
        class="nav"
      >
        <a
          aria-current="page"
          class="navItem active"
          href="/"
        >
          <div
            class="iconContainer"
          >
            <svg
              class="css-zqj3wo"
              viewBox="0 0 24 24"
            >
              <path
                d="M0 0h24v24H0V0z"
                fill="none"
              />
              <path
                d="M4 13h6c.55 0 1-.45 1-1V4c0-.55-.45-1-1-1H4c-.55 0-1 .45-1 1v8c0 .55.45 1 1 1zm0 8h6c.55 0 1-.45 1-1v-4c0-.55-.45-1-1-1H4c-.55 0-1 .45-1 1v4c0 .55.45 1 1 1zm10 0h6c.55 0 1-.45 1-1v-8c0-.55-.45-1-1-1h-6c-.55 0-1 .45-1 1v8c0 .55.45 1 1 1zM13 4v4c0 .55.45 1 1 1h6c.55 0 1-.45 1-1V4c0-.55-.45-1-1-1h-6c-.55 0-1 .45-1 1z"
              />
            </svg>
          </div>
          Dashboard
        </a>
        <a
          class="navItem"
          href="/flights"
        >
          <div
            class="iconContainer"
          >
            <svg
              class="css-zqj3wo"
              viewBox="0 0 24 24"
            >
              <g>
                <path
                  d="M0,0h24v24H0V0z"
                  fill="none"
                />
              </g>
              <g>
                <path
                  d="M20.19,4H4C2.9,4,2.01,4.9,2.01,6v4C3.11,10,4,10.9,4,12s-0.89,2-2,2v4c0,1.1,0.9,2,2,2h16c1.1,0,2-0.9,2-2V6 C22,4.9,21.19,4,20.19,4z M17.73,13.3l-8.49,2.26c-0.22,0.06-0.45-0.04-0.56-0.23l-1.12-1.95c-0.18-0.3-0.01-0.69,0.32-0.78h0 c0.16-0.04,0.34-0.01,0.47,0.1l1.05,0.82l2.39-0.64L9.9,9.6c-0.26-0.44-0.02-1.01,0.47-1.15l0,0c0.26-0.07,0.54,0,0.74,0.18 l3.69,3.44l2.44-0.65c0.51-0.14,1.04,0.17,1.18,0.68C18.55,12.62,18.25,13.15,17.73,13.3z"
                />
              </g>
            </svg>
          </div>
          Flights
        </a>
        <a
          class="navItem"
          href="/airports"
        >
          <div
            class="iconContainer"
          >
            <svg
              class="css-zqj3wo"
              viewBox="0 0 24 24"
            >
              <g>
                <rect
                  fill="none"
                  height="24"
                  width="24"
                  y="0"
                />
              </g>
              <g>
                <path
                  d="M12,2c-4.2,0-8,3.22-8,8.2c0,3.18,2.45,6.92,7.34,11.23c0.38,0.33,0.95,0.33,1.33,0 C17.55,17.12,20,13.38,20,10.2C20,5.22,16.2,2,12,2z M12,12c-1.1,0-2-0.9-2-2c0-1.1,0.9-2,2-2c1.1,0,2,0.9,2,2 C14,11.1,13.1,12,12,12z"
                />
              </g>
            </svg>
          </div>
          Airports
        </a>
        <a
          class="navItem"
          href="/aircraft"
        >
          <div
            class="iconContainer"
          >
            <svg
              class="css-zqj3wo"
              viewBox="0 0 24 24"
            >
              <path
                d="M0 0h24v24H0V0z"
                fill="none"
              />
              <path
                d="M21 14.58c0-.36-.19-.69-.49-.89L13 9V3.5c0-.83-.67-1.5-1.5-1.5S10 2.67 10 3.5V9l-7.51 4.69c-.3.19-.49.53-.49.89 0 .7.68 1.21 1.36 1L10 13.5V19l-1.8 1.35c-.13.09-.2.24-.2.4v.59c0 .33.32.57.64.48L11.5 21l2.86.82c.32.09.64-.15.64-.48v-.59c0-.16-.07-.31-.2-.4L13 19v-5.5l6.64 2.08c.68.21 1.36-.3 1.36-1z"
              />
            </svg>
          </div>
          Aircraft
        </a>
        <a
          class="navItem"
          href="/airlines"
        >
          <div
            class="iconContainer"
          >
            <svg
              class="css-zqj3wo"
              viewBox="0 0 24 24"
            >
              <rect
                fill="none"
                height="24"
                width="24"
                y="0"
              />
              <path
                d="M19.59,4h-5.01c-0.99,0-1.91,0.49-2.47,1.3L2,20h17l2.56-13.63C21.79,5.14,20.84,4,19.59,4z M14.5,14 c-1.38,0-2.5-1.12-2.5-2.5c0-1.38,1.12-2.5,2.5-2.5s2.5,1.12,2.5,2.5C17,12.88,15.88,14,14.5,14z"
              />
            </svg>
          </div>
          Airlines
        </a>
      </nav>
    </div>
    <div
      class="css-11duela"
    >
      <header
        class="css-abbpvh"
      >
        <h1
          class="css-1cbqxq6"
          data-testid="page-title"
        >
          Flights
        </h1>
      </header>
      <main
        class="css-1nkno5c"
      >
        <section
          class="css-1lb05ph"
        >
          <h2
            class="css-omoex7"
          >
            Flights
          </h2>
          <div
            class="tableContainer"
          >
            <table
              class="table"
            >
              <thead>
                <tr>
                  <th
                    class="tableHeader"
                  >
                    <button
                      class="sortButton"
                      type="button"
                    >
                      date
                      <div
                        class="css-1bnmoa1"
                      >
                        <span
                          class="sortable"
                        >
                          <svg
                            class="css-zqj3wo"
                            viewBox="0 0 24 24"
                          >
                            <path
                              d="M0 0h24v24H0V0z"
                              fill="none"
                            />
                            <path
                              d="M4 18h4c.55 0 1-.45 1-1s-.45-1-1-1H4c-.55 0-1 .45-1 1s.45 1 1 1zM3 7c0 .55.45 1 1 1h16c.55 0 1-.45 1-1s-.45-1-1-1H4c-.55 0-1 .45-1 1zm1 6h10c.55 0 1-.45 1-1s-.45-1-1-1H4c-.55 0-1 .45-1 1s.45 1 1 1z"
                            />
                          </svg>
                        </span>
                      </div>
                    </button>
                  </th>
                  <th
                    class="tableHeader"
                  >
                    <button
                      class="sortButton"
                      type="button"
                    >
                      source
                      <div
                        class="css-1bnmoa1"
                      >
                        <span
                          class="sortable"
                        >
                          <svg
                            class="css-zqj3wo"
                            viewBox="0 0 24 24"
                          >
                            <path
                              d="M0 0h24v24H0V0z"
                              fill="none"
                            />
                            <path
                              d="M4 18h4c.55 0 1-.45 1-1s-.45-1-1-1H4c-.55 0-1 .45-1 1s.45 1 1 1zM3 7c0 .55.45 1 1 1h16c.55 0 1-.45 1-1s-.45-1-1-1H4c-.55 0-1 .45-1 1zm1 6h10c.55 0 1-.45 1-1s-.45-1-1-1H4c-.55 0-1 .45-1 1s.45 1 1 1z"
                            />
                          </svg>
                        </span>
                      </div>
                    </button>
                  </th>
                  <th
                    class="tableHeader"
                  >
                    <button
                      class="sortButton"
                      type="button"
                    >
                      destination
                      <div
                        class="css-1bnmoa1"
                      >
                        <span
                          class="sortable"
                        >
                          <svg
                            class="css-zqj3wo"
                            viewBox="0 0 24 24"
                          >
                            <path
                              d="M0 0h24v24H0V0z"
                              fill="none"
                            />
                            <path
                              d="M4 18h4c.55 0 1-.45 1-1s-.45-1-1-1H4c-.55 0-1 .45-1 1s.45 1 1 1zM3 7c0 .55.45 1 1 1h16c.55 0 1-.45 1-1s-.45-1-1-1H4c-.55 0-1 .45-1 1zm1 6h10c.55 0 1-.45 1-1s-.45-1-1-1H4c-.55 0-1 .45-1 1s.45 1 1 1z"
                            />
                          </svg>
                        </span>
                      </div>
                    </button>
                  </th>
                  <th
                    class="tableHeader"
                  >
                    <button
                      class="sortButton"
                      type="button"
                    >
                      aircraft
                      <div
                        class="css-1bnmoa1"
                      >
                        <span
                          class="sortable"
                        >
                          <svg
                            class="css-zqj3wo"
                            viewBox="0 0 24 24"
                          >
                            <path
                              d="M0 0h24v24H0V0z"
                              fill="none"
                            />
                            <path
                              d="M4 18h4c.55 0 1-.45 1-1s-.45-1-1-1H4c-.55 0-1 .45-1 1s.45 1 1 1zM3 7c0 .55.45 1 1 1h16c.55 0 1-.45 1-1s-.45-1-1-1H4c-.55 0-1 .45-1 1zm1 6h10c.55 0 1-.45 1-1s-.45-1-1-1H4c-.55 0-1 .45-1 1s.45 1 1 1z"
                            />
                          </svg>
                        </span>
                      </div>
                    </button>
                  </th>
                  <th
                    class="tableHeader"
                  >
                    <button
                      class="sortButton"
                      type="button"
                    >
                      airline
                      <div
                        class="css-1bnmoa1"
                      >
                        <span
                          class="sortable"
                        >
                          <svg
                            class="css-zqj3wo"
                            viewBox="0 0 24 24"
                          >
                            <path
                              d="M0 0h24v24H0V0z"
                              fill="none"
                            />
                            <path
                              d="M4 18h4c.55 0 1-.45 1-1s-.45-1-1-1H4c-.55 0-1 .45-1 1s.45 1 1 1zM3 7c0 .55.45 1 1 1h16c.55 0 1-.45 1-1s-.45-1-1-1H4c-.55 0-1 .45-1 1zm1 6h10c.55 0 1-.45 1-1s-.45-1-1-1H4c-.55 0-1 .45-1 1s.45 1 1 1z"
                            />
                          </svg>
                        </span>
                      </div>
                    </button>
                  </th>
                </tr>
              </thead>
              <tbody
                data-testid="table-body"
              >
                <tr
                  class="tableRow"
                >
                  <td
                    class="textCell"
                  >
                    12/05/1965
                  </td>
                  <td
                    class="linkCell"
                  >
                    <a
                      href="/api/aircraft/3"
                    >
                      Bristol
                    </a>
                  </td>
                  <td
                    class="linkCell"
                  >
                    <a
                      href="/api/aircraft/2"
                    >
                      London Gatwick
                    </a>
                  </td>
                  <td
                    class="linkCell"
                  >
                    <a
                      href="/api/aircraft/2"
                    >
                      VP-BDN
                    </a>
                  </td>
                  <td
                    class="linkCell"
                  >
                    <a
                      href="/api/aircraft/2"
                    >
                      EasyJet
                    </a>
                  </td>
                </tr>
                <tr
                  class="tableRow"
                >
                  <td
                    class="textCell"
                  >
                    09/08/2010
                  </td>
                  <td
                    class="linkCell"
                  >
                    <a
                      href="/api/aircraft/3"
                    >
                      Bristol
                    </a>
                  </td>
                  <td
                    class="linkCell"
                  >
                    <a
                      href="/api/aircraft/2"
                    >
                      London Gatwick
                    </a>
                  </td>
                  <td
                    class="linkCell"
                  >
                    <a
                      href="/api/aircraft/3"
                    >
                      D-ABYS
                    </a>
                  </td>
                  <td
                    class="linkCell"
                  >
                    <a
                      href="/api/aircraft/2"
                    >
                      EasyJet
                    </a>
                  </td>
                </tr>
                <tr
                  class="tableRow"
                >
                  <td
                    class="textCell"
                  >
                    10/11/2020
                  </td>
                  <td
                    class="linkCell"
                  >
                    <a
                      href="/api/aircraft/3"
                    >
                      Bristol
                    </a>
                  </td>
                  <td
                    class="linkCell"
                  >
                    <a
                      href="/api/aircraft/2"
                    >
                      London Gatwick
                    </a>
                  </td>
                  <td
                    class="linkCell"
                  >
                    <a
                      href="/api/aircraft/3"
                    >
                      D-ABYS
                    </a>
                  </td>
                  <td
                    class="linkCell"
                  >
                    <a
                      href="/api/aircraft/2"
                    >
                      EasyJet
                    </a>
                  </td>
                </tr>
                <tr
                  class="tableRow"
                >
                  <td
                    class="textCell"
                  >
                    09/04/2014
                  </td>
                  <td
                    class="linkCell"
                  >
                    <a
                      href="/api/aircraft/3"
                    >
                      Bristol
                    </a>
                  </td>
                  <td
                    class="linkCell"
                  >
                    <a
                      href="/api/aircraft/2"
                    >
                      London Gatwick
                    </a>
                  </td>
                  <td
                    class="linkCell"
                  >
                    <a
                      href="/api/aircraft/3"
                    >
                      D-ABYS
                    </a>
                  </td>
                  <td
                    class="linkCell"
                  >
                    <a
                      href="/api/aircraft/2"
                    >
                      EasyJet
                    </a>
                  </td>
                </tr>
                <tr
                  class="tableRow"
                >
                  <td
                    class="textCell"
                  >
                    12/09/2014
                  </td>
                  <td
                    class="linkCell"
                  >
                    <a
                      href="/api/aircraft/2"
                    >
                      London Gatwick
                    </a>
                  </td>
                  <td
                    class="linkCell"
                  >
                    <a
                      href="/api/aircraft/3"
                    >
                      Bristol
                    </a>
                  </td>
                  <td
                    class="linkCell"
                  >
                    <a
                      href="/api/aircraft/3"
                    >
                      D-ABYS
                    </a>
                  </td>
                  <td
                    class="linkCell"
                  >
                    <a
                      href="/api/aircraft/3"
                    >
                      Virgin Atlantic
                    </a>
                  </td>
                </tr>
                <tr
                  class="tableRow"
                >
                  <td
                    class="textCell"
                  >
                    30/08/1962
                  </td>
                  <td
                    class="linkCell"
                  >
                    <a
                      href="/api/aircraft/2"
                    >
                      London Gatwick
                    </a>
                  </td>
                  <td
                    class="linkCell"
                  >
                    <a
                      href="/api/aircraft/3"
                    >
                      Bristol
                    </a>
                  </td>
                  <td
                    class="linkCell"
                  >
                    <a
                      href="/api/aircraft/3"
                    >
                      D-ABYS
                    </a>
                  </td>
                  <td
                    class="linkCell"
                  >
                    <a
                      href="/api/aircraft/3"
                    >
                      Virgin Atlantic
                    </a>
                  </td>
                </tr>
                <tr
                  class="tableRow"
                >
                  <td
                    class="textCell"
                  >
                    11/09/2011
                  </td>
                  <td
                    class="linkCell"
                  >
                    <a
                      href="/api/aircraft/3"
                    >
                      Bristol
                    </a>
                  </td>
                  <td
                    class="linkCell"
                  >
                    <a
                      href="/api/aircraft/3"
                    >
                      Bristol
                    </a>
                  </td>
                  <td
                    class="linkCell"
                  >
                    <a
                      href="/api/aircraft/2"
                    >
                      VP-BDN
                    </a>
                  </td>
                  <td
                    class="linkCell"
                  >
                    <a
                      href="/api/aircraft/2"
                    >
                      EasyJet
                    </a>
                  </td>
                </tr>
                <tr
                  class="tableRow"
                >
                  <td
                    class="textCell"
                  >
                    03/01/2004
                  </td>
                  <td
                    class="linkCell"
                  >
                    <a
                      href="/api/aircraft/3"
                    >
                      Bristol
                    </a>
                  </td>
                  <td
                    class="linkCell"
                  >
                    <a
                      href="/api/aircraft/2"
                    >
                      London Gatwick
                    </a>
                  </td>
                  <td
                    class="linkCell"
                  >
                    <a
                      href="/api/aircraft/2"
                    >
                      VP-BDN
                    </a>
                  </td>
                  <td
                    class="linkCell"
                  >
                    <a
                      href="/api/aircraft/3"
                    >
                      Virgin Atlantic
                    </a>
                  </td>
                </tr>
                <tr
                  class="tableRow"
                >
                  <td
                    class="textCell"
                  >
                    04/12/2009
                  </td>
                  <td
                    class="linkCell"
                  >
                    <a
                      href="/api/aircraft/3"
                    >
                      Bristol
                    </a>
                  </td>
                  <td
                    class="linkCell"
                  >
                    <a
                      href="/api/aircraft/3"
                    >
                      Bristol
                    </a>
                  </td>
                  <td
                    class="linkCell"
                  >
                    <a
                      href="/api/aircraft/2"
                    >
                      VP-BDN
                    </a>
                  </td>
                  <td
                    class="linkCell"
                  >
                    <a
                      href="/api/aircraft/2"
                    >
                      EasyJet
                    </a>
                  </td>
                </tr>
                <tr
                  class="tableRow"
                >
                  <td
                    class="textCell"
                  >
                    13/03/2013
                  </td>
                  <td
                    class="linkCell"
                  >
                    <a
                      href="/api/aircraft/2"
                    >
                      London Gatwick
                    </a>
                  </td>
                  <td
                    class="linkCell"
                  >
                    <a
                      href="/api/aircraft/3"
                    >
                      Bristol
                    </a>
                  </td>
                  <td
                    class="linkCell"
                  >
                    <a
                      href="/api/aircraft/2"
                    >
                      VP-BDN
                    </a>
                  </td>
                  <td
                    class="linkCell"
                  >
                    <a
                      href="/api/aircraft/2"
                    >
                      EasyJet
                    </a>
                  </td>
                </tr>
              </tbody>
            </table>
          </div>
          <div
            class="pagination"
          >
            <div
              aria-labelledby="page-size-controls-legend-1"
              class="fieldset"
              role="group"
            >
              <span
                class="legend"
              >
                Showing
              </span>
              <div
                class="inputsContainer"
              >
                <input
                  checked=""
                  class="input"
                  id="page-size-option-2"
                  name="page-size"
                  type="radio"
                  value="10"
                />
                <label
                  class="label"
                  for="page-size-option-2"
                >
                  10
                   
                  <span
                    class="visuallyHidden"
                  >
                    results per page
                  </span>
                </label>
                <input
                  class="input"
                  id="page-size-option-3"
                  name="page-size"
                  type="radio"
                  value="20"
                />
                <label
                  class="label"
                  for="page-size-option-3"
                >
                  20
                   
                  <span
                    class="visuallyHidden"
                  >
                    results per page
                  </span>
                </label>
                <input
                  class="input"
                  id="page-size-option-4"
                  name="page-size"
                  type="radio"
                  value="50"
                />
                <label
                  class="label"
                  for="page-size-option-4"
                >
                  50
                   
                  <span
                    class="visuallyHidden"
                  >
                    results per page
                  </span>
                </label>
              </div>
              <span
                class="legend"
                id="page-size-controls-legend-1"
              >
                results per page
              </span>
            </div>
            <div
              class="pageControlsContainer"
              data-testid="page-controls"
            >
              <button
                aria-label="First Page"
                class="button"
                data-testid="first-page-button"
                disabled=""
                type="button"
              >
                <svg
                  class="css-zqj3wo"
                  viewBox="0 0 24 24"
                >
                  <path
                    d="M24 0v24H0V0h24z"
                    fill="none"
                    opacity=".87"
                  />
                  <path
                    d="M17.7 15.89L13.82 12l3.89-3.89c.39-.39.39-1.02 0-1.41-.39-.39-1.02-.39-1.41 0l-4.59 4.59c-.39.39-.39 1.02 0 1.41l4.59 4.59c.39.39 1.02.39 1.41 0 .38-.38.38-1.02-.01-1.4zM7 6c.55 0 1 .45 1 1v10c0 .55-.45 1-1 1s-1-.45-1-1V7c0-.55.45-1 1-1z"
                  />
                </svg>
              </button>
              <button
                aria-label="Previous Page"
                class="button"
                data-testid="previous-page-button"
                disabled=""
                type="button"
              >
                <svg
                  class="css-zqj3wo"
                  viewBox="0 0 24 24"
                >
                  <path
                    d="M0 0h24v24H0V0z"
                    fill="none"
                  />
                  <path
                    d="M14.91 6.71c-.39-.39-1.02-.39-1.41 0L8.91 11.3c-.39.39-.39 1.02 0 1.41l4.59 4.59c.39.39 1.02.39 1.41 0 .39-.39.39-1.02 0-1.41L11.03 12l3.88-3.88c.38-.39.38-1.03 0-1.41z"
                  />
                </svg>
              </button>
              <span
                class="currentPage"
              >
                Page 
                1
                 of 
                1
              </span>
              <button
                aria-label="Next Page"
                class="button"
                data-testid="next-page-button"
                disabled=""
                type="button"
              >
                <svg
                  class="css-zqj3wo"
                  viewBox="0 0 24 24"
                >
                  <path
                    d="M0 0h24v24H0V0z"
                    fill="none"
                  />
                  <path
                    d="M9.31 6.71c-.39.39-.39 1.02 0 1.41L13.19 12l-3.88 3.88c-.39.39-.39 1.02 0 1.41.39.39 1.02.39 1.41 0l4.59-4.59c.39-.39.39-1.02 0-1.41L10.72 6.7c-.38-.38-1.02-.38-1.41.01z"
                  />
                </svg>
              </button>
              <button
                aria-label="Last Page"
                class="button"
                data-testid="last-page-button"
                disabled=""
                type="button"
              >
                <svg
                  class="css-zqj3wo"
                  viewBox="0 0 24 24"
                >
                  <path
                    d="M0 0h24v24H0V0z"
                    fill="none"
                    opacity=".87"
                  />
                  <path
                    d="M6.29 8.11L10.18 12l-3.89 3.89c-.39.39-.39 1.02 0 1.41.39.39 1.02.39 1.41 0l4.59-4.59c.39-.39.39-1.02 0-1.41L7.7 6.7c-.39-.39-1.02-.39-1.41 0-.38.39-.38 1.03 0 1.41zM17 6c.55 0 1 .45 1 1v10c0 .55-.45 1-1 1s-1-.45-1-1V7c0-.55.45-1 1-1z"
                  />
                </svg>
              </button>
            </div>
          </div>
        </section>
      </main>
    </div>
  </div>
</div>
`;
